pkg/apps/blackduck/components/rc/cfssl/v1/component.go

```diff
@@ -96,6 +97,7 @@ func (c *BdReplicationController) getCfsslolumeMounts() []*horizonapi.VolumeMoun
 	return volumesMounts
 }
 
-func NewBdReplicationController(replicationController *types.ReplicationController, config *protoform.Config, kubeClient *kubernetes.Clientset, blackduck *blackduckapi.Blackduck) types.ReplicationControllerInterface {
-	return &BdReplicationController{ReplicationController: replicationController, config: config, kubeClient: kubeClient, blackduck: blackduck}
+// NewBdReplicationController returns the Black Duck RC configuration
+func NewBdReplicationController(replicationController *types.ReplicationController, config *protoform.Config, kubeClient *kubernetes.Clientset, blackDuck interface{}) types.ReplicationControllerInterface {
+	return &BdReplicationController{ReplicationController: replicationController, config: config, kubeClient: kubeClient, blackDuck: blackDuck.(*blackduckapi.Blackduck)}
```

Comment:
Should cast the blackDuck instance and verify it is the correct type, return error otherwise.  Blindly casting here (and other similar functions) opens up crash scenarios.

Reply:
fixed